qiskit/transpiler/passes/optimization/collect_2q_blocks.py

```diff
@@ -19,232 +19,61 @@
 
 
 class Collect2qBlocks(AnalysisPass):
-    """Collect sequences of uninterrupted gates acting on 2 qubits.
-
-    Traverse the DAG and find blocks of gates that act consecutively on
-    pairs of qubits. Write the blocks to property_set as a dictionary
-    of the form::
-
-        {(q0, q1): [[g0, g1, g2], [g5]],
-         (q0, q2): [[g3, g4]]
-         ..
-         .
-        }
-
-    Based on implementation by Andrew Cross.
-    """
+    """Collect two-qubit subcircuits."""
 
     def run(self, dag):
         """Run the Collect2qBlocks pass on `dag`.
 
-        The blocks contain "op" nodes in topological sort order
-        such that all gates in a block act on the same pair of
-        qubits and are adjacent in the circuit. the blocks are built
-        by examining predecessors and successors of 2q gates in
-        the circuit.
+        The blocks contain "op" nodes in topological order such that all gates
+        in a block act on the same qubits and are adjacent in the circuit.
 
-        After the execution, ``property_set['block_list']`` is set to
-        a list of tuples of "op" node labels.
+        After the execution, ``property_set['block_list']`` is set to a list of
+        tuples of "op" node.
         """
-        # Initiate the commutation set
         self.property_set["commutation_set"] = defaultdict(list)
-
-        block_list = []
-        nodes = list(dag.topological_nodes())
-        nodes_seen = dict(zip(nodes, [False] * len(nodes)))
-        for nd in dag.topological_op_nodes():
-
-            group = []
-            # Explore predecessors and successors of 2q gates
-            if (  # pylint: disable=too-many-boolean-expressions
-                nd.type == "op"
-                and isinstance(nd._op, Gate)
-                and len(nd._qargs) == 2
-                and not nodes_seen[nd]
-                and nd._op.condition is None
-                and not nd._op.is_parameterized()
+        pending_1q = [list()] * dag.num_qubits()
```

Comment:
```suggestion
        pending_1q = [list() for _ in range(dag.num_qubits())]
```
otherwise each entry of `pending_1q` will be the *same* list and cases like
```python
qc = QuantumCircuit(3)
qc.x(0)
qc.cx(1, 2)
```
will give wrong output.